### ISSUE TITLE
[RFC] Warn if image from another registry is used in `FROM`

### ISSUE THREAD

**Issue by lorenzo** [hadolint/hadolint#225]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

This feature is actually 2 features in one. The first feature is about automatically checking that a Dockerfile is does not use images from untrusted repositories.

Using the `FROM` instruction is a huge exercise in trust, you have to trust that a particular version or an image is safe for you to use, and that it will never be retagged maliciously. In order to prevent that, some companies copies trusted images into their own repositories, and reference them directly.

For example, this would be an untrusted image:

```dockerfile
FROM randomguy/fancy:10
...
```

But after an audit, the company decides to copy the image into their own repository, as it was deemed safe:

```dockerfile
FROM my-registry.com/fancy:10
...
```

The idea is that hadolint can warn whenever an untrusted repo is used. I propose the new `--trusted-repos` flag:

```bash
hadolint --trusted-repos my-registry.com Dockerfile
```

In order to support this feature, another feature needs to be implemented first: Optional rules. These type of rules won't run by default unless enabled by a flag or by a configuration file. The configuration file for this feature could for example be:

```yaml
ignored:
  - DL3002
trusted_repos:
  - my-registry.com
```

@zemanlx do you think hadolint is a good place to have this feature?

**Comment by zemanlx** [comment]
That sounds great! Do you want to have tags/labels on rules to know which option should trigger them?

Regarding optional/categories rules. It would be useful for CI purposes if we are able to set also strictness of `hadolint` to exit with an error (non `0`) only in case of (error) or (error, warning) or (error, warning, info) in loglevel fashion.

**Comment by lorenzo** [comment]
@zemanlx since I only have an use case for the optional rules, I was thinking to have just a separate list. If we need anything more fancy later, I can look into adding tags to them.

> if we are able to set also strictness of hadolint

I've been also thinking about it and would like to implement it, but currently it seems to me that the severity of certain rules was not thought through in some cases. Would you like to review those severity fields and see if they make sense as they are?
